packages/core/ui-settings/core-ui-settings-server-internal/src/routes/set.ts

```diff
@@ -22,36 +26,55 @@ const validate = {
 };
 
 export function registerSetRoute(router: InternalUiSettingsRouter) {
-  router.post(
-    { path: '/api/kibana/settings/{key}', validate },
-    async (context, request, response) => {
-      try {
-        const uiSettingsClient = (await context.core).uiSettings.client;
+  const setFromRequest = async (
+    uiSettingsClient: IUiSettingsClient,
+    context: InternalUiSettingsRequestHandlerContext,
+    request: KibanaRequest<
+      Readonly<{} & { key: string }>,
+      unknown,
+      Readonly<{ value?: any } & {}>,
+      'post'
+    >,
+    response: KibanaResponseFactory
+  ) => {
+    try {
+      const { key } = request.params;
+      const { value } = request.body;
 
-        const { key } = request.params;
-        const { value } = request.body;
+      await uiSettingsClient.set(key, value);
 
-        await uiSettingsClient.set(key, value);
-
-        return response.ok({
-          body: {
-            settings: await uiSettingsClient.getUserProvided(),
-          },
+      return response.ok({
+        body: {
+          settings: await uiSettingsClient.getUserProvided(),
+        },
+      });
+    } catch (error) {
+      if (SavedObjectsErrorHelpers.isSavedObjectsClientError(error)) {
+        return response.customError({
+          body: error,
+          statusCode: error.output.statusCode,
         });
-      } catch (error) {
-        if (SavedObjectsErrorHelpers.isSavedObjectsClientError(error)) {
-          return response.customError({
-            body: error,
-            statusCode: error.output.statusCode,
-          });
-        }
-
-        if (error instanceof CannotOverrideError || error instanceof ValidationError) {
-          return response.badRequest({ body: error });
-        }
+      }
 
-        throw error;
+      if (error instanceof CannotOverrideError || error instanceof ValidationError) {
+        return response.badRequest({ body: error });
       }
+
+      throw error;
+    }
+  };
+  router.post(
+    { path: '/api/kibana/settings/{key}', validate },
```

Comment:
NIT: we have a path conflict when trying to update a setting named `global` ( between the old settings' `set` route - `POST '/api/kibana/settings/{key}'` and the global settings' `set_many` route- `POST '/api/kibana/settings/global'`). 

It's honestly fine, the edge case is acceptable, so nothing blocking. I still wonder if it wouldn't be safer to go with a distinct path prefix, e.g `/api/kibana/global_settings` instead of `/api/kibana/settings/global`?

Reply:
Was thinking about that too; `global_settings` may be a better option eventually